pkg/sink/codec/debezium/codec.go

```diff
@@ -0,0 +1,408 @@
+				col.Value,
+				col.Name))
+	case mysql.TypeTimestamp:
+		// Debezium behavior from doc:
+		// > The TIMESTAMP type represents a timestamp without time zone information.
+		// > It is converted by MySQL from the server (or session's) current time zone into UTC
+		// > when writing and from UTC into the server (or session's) current time zone when reading
+		// > back the value.
+		// > Such columns are converted into an equivalent io.debezium.time.ZonedTimestamp in UTC
+		// > based on the server (or session's) current time zone. The time zone will be queried from
+		// > the server by default. If this fails, it must be specified explicitly by the database
+		// > connectionTimeZone MySQL configuration option.
+		if v, ok := col.Value.(string); ok {
+			t, err := time.ParseInLocation("2006-01-02 15:04:05.999999", v, c.config.TimeZone)
+			if err != nil {
+				// For example, time may be invalid like 1000-00-00
+				if mysql.HasNotNullFlag(ft.GetFlag()) {
+					t = time.Unix(0, 0)
+				} else {
+					return nil, nil
+				}
+			}
+
+			str := t.UTC().Format("2006-01-02T15:04:05")
+			fsp := ft.GetDecimal()
+			if fsp > 0 {
+				tmp := fmt.Sprintf(".%06d", t.Nanosecond()/1000)
+				str = str + tmp[:1+fsp]
+			}
+			str += "Z"
+
+			return str, nil
+		}
+		return nil, cerror.WrapError(
+			cerror.ErrDebeziumEncodeFailed,
+			errors.Errorf(
+				"unexpected column value type %T for timestamp column %s",
+				col.Value,
+				col.Name))
+	case mysql.TypeDuration:
+		// Debezium behavior from doc:
+		// > Represents the time value in microseconds and does not include
+		// > time zone information. MySQL allows M to be in the range of 0-6.
+		if v, ok := col.Value.(string); ok {
+			ctx := &stmtctx.StatementContext{}
+			d, _, _, err := types.StrToDuration(ctx, v, ft.GetDecimal())
+			if err != nil {
+				return nil, cerror.WrapError(
+					cerror.ErrDebeziumEncodeFailed,
+					err)
+			}
+			return d.Microseconds(), nil
+		}
+		return nil, cerror.WrapError(
+			cerror.ErrDebeziumEncodeFailed,
+			errors.Errorf(
+				"unexpected column value type %T for time column %s",
+				col.Value,
+				col.Name))
+	case mysql.TypeLonglong:
+		if col.Flag.IsUnsigned() {
+			// Handle with BIGINT UNSIGNED.
+			// Debezium always produce INT64 instead of UINT64 for BIGINT.
+			if v, ok := col.Value.(uint64); ok {
+				return int64(v), nil
+			}
+			return nil, cerror.WrapError(
+				cerror.ErrDebeziumEncodeFailed,
+				errors.Errorf(
+					"unexpected column value type %T for unsigned bigint column %s",
+					col.Value,
+					col.Name))
+		}
+
+		// Note: Although Debezium's doc claims to use INT32 for INT, but it
+		// actually uses INT64. Debezium also uses INT32 for SMALLINT.
+		// So we only handle with TypeLonglong here.
+	}
+	return col.Value, nil
+}
+
+func (c *Codec) rowChangeToDebeziumMsg(e *model.RowChangedEvent) (*debeziumDataChangeMsg, error) {
+	commitTime, _ := tsoutil.ParseTS(e.CommitTs)
+
+	source := &debeziumMsgSource{
+		Version:   "2.4.0.Final",
```

Comment:
how about make this value constant, and add some comment about it.

Reply:
I'm not sure about this. As you see, if we introduce a constant elsewhere, we need to supply comment to describe what the constant is. While in the current impl, it is very clear what is it.